### ISSUE TITLE
[Bug] v0.14 does not work properly

### ISSUE THREAD

**Issue by niuiic** [farm-fe/farm#747]
### Steps to reproduce

```sh
git clone https://github.com/niuiic/fx-flow -b feat/pack-in-chunks
pnpm initialize
pnpm build
```

Copy `packages/core/dist` to another project and write a test file.

```
mkdir test
cd test
pnpm init
pnpm add fx-flow
# replace node_modules/fx-flow/dist with the built one
```

`index.mjs`
```typescript
import { add } from "fx-flow";

console.log(add(1, 2));
```

Continued from #741. I tried v0.14 and new issues were found.

### What is expected?

output 3

### What is actually happening?

```
ReferenceError: location is not defined
    at Object.moduleCreated (file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs:1:8023)
    at i.hookSerial (file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs:1:3817)
    at d.require (file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs:1:1788)
    at file:///home/niuiic/Downloads/test/node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs:1:5350
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)

Node.js v21.1.0
```

Another issue, run `pnpm build --filter fx-flow --force` two times.

```
fx-flow:build: cache bypass, force executing 944fccf30bc0be06
fx-flow:build:
fx-flow:build: > fx-flow@0.4.2 build /home/niuiic/Documents/projects/node/fx-flow/packages/core
fx-flow:build: > farm build && farm build -c farm.config.cjs.ts
fx-flow:build:
fx-flow:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/node/fx-flow/packages/core/farm.config.ts
fx-flow:build: thread '<unnamed>' panicked at crates/plugin_script/src/handle_entry_resources.rs:409:8:
fx-flow:build: runtime resource is not found
fx-flow:build: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fx-flow:build:  ELIFECYCLE  Command failed with exit code 134.
fx-flow:build: ERROR: command finished with error: command (/home/niuiic/Documents/projects/node/fx-flow/packages/core) pnpm run build exited (1)
fx-flow#build: command (/home/niuiic/Documents/projects/node/fx-flow/packages/core) pnpm run build exited (1)

 Tasks:    0 successful, 1 total
Cached:    0 cached, 1 total
  Time:    3.781s
Failed:    fx-flow#build

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
```

### System Info

```Shell
System:
    OS: Linux 6.1 Gentoo Linux
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 13.48 GB / 15.59 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.1.0 - /run/user/1000/fnm_multishells/7878_1700441074872/bin/node
    npm: 10.2.0 - /run/user/1000/fnm_multishells/7878_1700441074872/bin/npm
    pnpm: 8.10.5 - /run/user/1000/fnm_multishells/7878_1700441074872/bin/pnpm
```
```


### Any additional comments?

_No response_

**Comment by wre232114** [comment]
I'll try. it should work, all farm packages are build with farm now and it works

**Comment by ErKeLost** [comment]
![image](https://github.com/farm-fe/farm/assets/66500121/8775f5ae-c2ee-43a2-9b19-5561ed78ea32)
Output configuration needs to be configured. TargetEnv is node

**Comment by ErKeLost** [comment]
But I found other problems with setting the cache to false, but the cache was generated anyway.

![image](https://github.com/farm-fe/farm/assets/66500121/9fe80ecb-22a1-4e2a-ad40-4b2c6bc2c043)

The second problem caching causes the second compilation to fail.

![image](https://github.com/farm-fe/farm/assets/66500121/31815a2e-5f48-41a3-9819-f759b73f54d9)

**Comment by niuiic** [comment]
> ![image](https://user-images.githubusercontent.com/66500121/284109808-8775f5ae-c2ee-43a2-9b19-5561ed78ea32.png) Output configuration needs to be configured. TargetEnv is node @niuiic

It seems that this option makes it work for nodejs, but not browser.

```
[ Farm ] Using config file at /home/niuiic/Documents/projects/gallery/apps/website/farm.config.ts
website:dev:
website:dev:  ϟ  Farm  v0.14.0
website:dev:  ✓  Ready in 221ms
website:dev:
website:dev: [ Farm ] ➡️ Local:   http://localhost:9000
website:dev: [ Farm ] ➡️ Network: http://10.10.113.123:9000
website:dev: [ Farm ] Error: Can not load `../../node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs`. Original error:
website:dev: Some(GenericError("Os { code: 2, kind: NotFound, message: \"No such file or directory\" }")).
website:dev:
website:dev: Potential Causes:
website:dev: 1.This kind of module is not supported, you may need plugins to support it.
website:dev:
website:dev: [ Farm ] Error: Can not load `../../node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs`. Original error:
website:dev: Some(GenericError("Os { code: 2, kind: NotFound, message: \"No such file or directory\" }")).
website:dev:
website:dev: Potential Causes:
website:dev: 1.This kind of module is not supported, you may need plugins to support it.
website:dev:
website:dev: [ Farm ] Error: Can not load `../../node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs`. Original error:
website:dev: Some(GenericError("Os { code: 2, kind: NotFound, message: \"No such file or directory\" }")).
website:dev:
website:dev: Potential Causes:
website:dev: 1.This kind of module is not supported, you may need plugins to support it.
website:dev:
website:dev: [ Farm ] Error: Can not load `../../node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/index.mjs`. Original error:
website:dev: Some(GenericError("Os { code: 2, kind: NotFound, message: \"No such file or directory\" }")).
website:dev:
website:dev: Potential Causes:
website:dev: 1.This kind of module is not supported, you may need plugins to support it.
```

**Comment by niuiic** [comment]
And a related feature request. Do you plan to support dynamic import?

**Comment by wre232114** [comment]
dynamic report already supported

**Comment by niuiic** [comment]
My mistake, I found it in the document.

**Comment by ErKeLost** [comment]
> > ![image](https://user-images.githubusercontent.com/66500121/284109808-8775f5ae-c2ee-43a2-9b19-5561ed78ea32.png) Output configuration needs to be configured. TargetEnv is node @niuiic
> 
> It seems that this option makes it work for nodejs, but not browser.
> 
> Can I build for both nodejs and browser? It's odd that a platform-independent library needs to be packaged separately for different platforms.
> 
> ```
> website:dev:
> website:dev:  Build failed due to errors:
> website:dev:
> website:dev:  Can not resolve `node:module` from ../../node_modules/.pnpm/fx-flow@0.4.2/node_modules/fx-flow/dist/esm/__farm_runtime.mjs.
> website:dev: Original error: None.
> website:dev:
> website:dev: Potential Causes:
> website:dev: 1.The file that `node:module` points to does not exist.
> website:dev: 2.Install it first if `node:module` is an dependency from node_modules, if you are using pnpm refer to [https://pnpm.io/faq#pnpm-does-not-work-with-your-project-here] for solutions.
> website:dev: 3. If `node:module` is a alias, make sure your alias config is correct.
> website:dev:
> website:dev:  ELIFECYCLE  Command failed with exit code 1.
> website:dev: ERROR: command finished with error: command (/home/niuiic/Documents/projects/gallery/apps/website) pnpm run dev exited (1)
> website#dev: command (/home/niuiic/Documents/projects/gallery/apps/website) pnpm run dev exited (1)
> ```

This problem will be solved in v0.14.1

**Comment by ErKeLost** [comment]
fixed in v0.14.1 @niuiic

**Comment by wre232114** [comment]
It still failed in v0.14.1

**Comment by wre232114** [comment]
fixed in #758
